whylog_vim/proxy/teacher/__init__.py

```diff
@@ -1,17 +1,60 @@
+from whylog_vim.consts import EditorStates, Messages
+from whylog_vim.output_formater.teacher_formater import TeacherFormater
+from whylog_vim.proxy.teacher.utils import get_next_parser_id
+from whylog_vim.proxy.teacher.exceptions import CannotGoToPosition
+
+
 class TeacherProxy(object):
     def __init__(self, teacher, editor, main_proxy):
         self.teacher = teacher
         self.editor = editor
         self.main_proxy = main_proxy
 
+    def new_lesson(self):
+        front_input = self.editor.get_front_input()
+        self.teacher.add_line(get_next_parser_id(), front_input, effect=True)
+        self.origin_file_name = self.editor.get_current_filename()
+        print(Messages.ADDED_EFFECT)
+
+    def add_cause(self):
+        front_input = self.editor.get_front_input()
+        self.teacher.add_line(get_next_parser_id(), front_input)
+        self.editor.create_teacher_window()
+        self.print_teacher()
+
     def handle_menu_signal(self):
-        pass
+        self._set_cursor_position()
+        self.output.call_button(self.editor.get_line_number())
+        if self.main_proxy.get_state() == EditorStates.TEACHER:
+            self._return_cursor_to_position()
+
+    def set_read_function(self, read_function):
```

Comment:
this method is essentially a setter. You don't need to define those in python, but can use `object.field = value` instead and make it a setter later if necessary.

From reading this class it is hard to understand what the `read_function` is for. Please document it.